src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1436 @@
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* Call the full RHS if needed. If this is the first step then we may need to
+     evaluate or copy the RHS values from an  earlier evaluation (e.g., to
+     compute h0). For subsequent steps treat this RHS evaluation as an
+     evaluation at the end of the just completed step to potentially reuse
+     (FSAL methods) RHS evaluations from the end of the last step. */
+
+  if (!(ark_mem->fn_is_current) && ark_mem->initsetup)
+  {
+    retval = step_mem->fe(ark_mem->tn, ark_mem->yn, ark_mem->fn,
+                          ark_mem->user_data);
+    step_mem->nfe++;
+    ark_mem->fn_is_current = SUNTRUE;
+    if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+  }
+
+  /* A tentative solution at t+h is returned in
+     y and its slope is evaluated in temp1.  */
+
+  N_VScale(ONE, ark_mem->yn, ark_mem->tempv2);
+
+  N_VLinearSum(ONE, ark_mem->yn, onesixth * ark_mem->h, ark_mem->fn,
+               ark_mem->ycur);
+  N_VLinearSum(ONE, ark_mem->yn, 1.0 / 5.0 * ark_mem->h, ark_mem->fn,
+               ark_mem->tempv1);
+
+  /* Evaluate stages j = 2,...,step_mem->reqstages */
+  for (int j = 2; j <= 5; j++)
+  {
+    retval = step_mem->fe(ark_mem->tcur +
+                            ((sunrealtype)j - ONE) * onesixth * ark_mem->h,
+                          ark_mem->ycur, ark_mem->fn, ark_mem->user_data);
+    if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+
+    N_VLinearSum(ONE, ark_mem->ycur, onesixth * ark_mem->h, ark_mem->fn,
+                 ark_mem->ycur);
+    if (j == 4)
+    {
+      N_VLinearSum(ONE, ark_mem->tempv1, 3.0 / 10.0 * ark_mem->h, ark_mem->fn,
+                   ark_mem->tempv1);
+    }
+  }
+  N_VLinearSum(1.0 / 25.0, ark_mem->tempv2, 9.0 / 25.0, ark_mem->ycur,
+               ark_mem->tempv2);
+  N_VLinearSum(15, ark_mem->tempv2, -5, ark_mem->ycur, ark_mem->ycur);
+  for (int j = 6; j <= 9; j++)
+  {
+    retval = step_mem->fe(ark_mem->tcur +
+                            ((sunrealtype)j - 4.0) * onesixth * ark_mem->h,
+                          ark_mem->ycur, ark_mem->fn, ark_mem->user_data);
+    if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+
+    N_VLinearSum(ONE, ark_mem->ycur, onesixth * ark_mem->h, ark_mem->fn,
+                 ark_mem->ycur);
+
+    if (j == 7)
+    {
+      N_VLinearSum(ONE, ark_mem->tempv1, 1.0 / 5.0 * ark_mem->h, ark_mem->fn,
+                   ark_mem->tempv1);
+    }
+    if (j == 9)
+    {
+      N_VLinearSum(ONE, ark_mem->tempv1, 3.0 / 10.0 * ark_mem->h, ark_mem->fn,
+                   ark_mem->tempv1);
+    }
+  }
+
+  retval = step_mem->fe(ark_mem->tcur + ark_mem->h, ark_mem->ycur, ark_mem->fn,
+                        ark_mem->user_data);
+  if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+
+  N_VLinearSum(ONE, ark_mem->tempv2, 3.0 / 5.0, ark_mem->ycur, ark_mem->ycur);
+  N_VLinearSum(ONE, ark_mem->ycur, 1.0 / 10.0 * ark_mem->h, ark_mem->fn,
+               ark_mem->ycur);
```

Comment:
Replace these two calls to `N_VLinearSum` with a single call to `N_VLinearCombination`, and use `SUN_RCONST` for all of these constant values.